scss/_patterns_accordion.scss

```diff
@@ -59,53 +61,93 @@
     }
   }
 
-  .p-accordion__title {
-    margin-bottom: 0;
-    padding-top: 0;
-    position: relative;
-    text-indent: -$sph-inner - $icon-size;
+  @include deprecate('3.0.0', '.p-accordion__title and .p-accordion__tab--with-title are being deprecated, please use .p-accordion__heading > .p-accordion__tab pattern instead') {
+    .p-accordion__title {
+      margin-bottom: 0;
+      padding-top: 0;
+      position: relative;
+      text-indent: -$sph-inner - $icon-size;
+
+      &::before {
+        background-size: $icon-size;
+        content: '';
+        display: inline-block;
+        height: $icon-size;
+        margin-right: $sph-inner;
+        width: $icon-size;
+      }
+    }
 
-    &::before {
-      background-size: $icon-size;
-      content: '';
-      display: inline-block;
-      height: $icon-size;
-      margin-right: $sph-inner;
-      width: $icon-size;
+    // FIXME: we shouldn't be targeting types directly here
+    // stylelint-disable selector-max-type
+    h2.p-accordion__title::before {
+      // make icon bigger for h2 (same as p-icon)
+      $icon-size-h2: $x-height;
+
+      background-size: $icon-size-h2;
+      height: $icon-size-h2;
+      width: $icon-size-h2;
     }
-  }
 
-  // FIXME: we shouldn't be targeting types directly here
-  // stylelint-disable selector-max-type
-  h2.p-accordion__title::before {
-    // make icon bigger for h2 (same as p-icon)
-    $icon-size-h2: $x-height;
+    h5.p-accordion__title::before,
+    h6.p-accordion__title::before {
+      // align icon within font metrics same as p-icon pattern
+      $vertical-offset: 0.5px;
 
-    background-size: $icon-size-h2;
-    height: $icon-size-h2;
-    width: $icon-size-h2;
-  }
+      vertical-align: calc(#{$vertical-offset} + #{0.5 * $cap-height} - #{0.5 * $icon-size});
+    }
+    // stylelint-enable selector-max-type
 
-  h5.p-accordion__title::before,
-  h6.p-accordion__title::before {
-    // align icon within font metrics same as p-icon pattern
-    $vertical-offset: 0.5px;
+    // override base expanded button styles
+    .p-accordion__tab--with-title[aria-expanded] {
+      background-color: inherit;
+    }
 
-    vertical-align: calc(#{$vertical-offset} + #{0.5 * $cap-height} - #{0.5 * $icon-size});
-  }
-  // stylelint-enable selector-max-type
+    .p-accordion__tab--with-title[aria-expanded='true'] .p-accordion__title::before {
+      @include vf-icon-minus($color-mid-dark);
+    }
 
-  // override base expanded button styles
-  .p-accordion__tab--with-title[aria-expanded] {
-    background-color: inherit;
+    .p-accordion__tab--with-title[aria-expanded='false'] .p-accordion__title::before {
+      @include vf-icon-plus($color-mid-dark);
+    }
   }
 
-  .p-accordion__tab--with-title[aria-expanded='true'] .p-accordion__title::before {
-    @include vf-icon-minus($color-mid-dark);
-  }
+  .p-accordion__heading {
+    margin-bottom: 0;
+    max-width: none;
+    padding-top: 0;
+
+    @at-root {
```

Comment:
While it is a valid use case for `@at-root`, it seems that there is no really much benefit of using that.
It doesn't make the code shorter and it doesn't make it easier to read (quite contrary, the combination of `@at-root` and `#{&}` requires quite a bit of understanding of SASS to process).

I think explicitly writing it as:

```scss
  // stylelint-disable selector-max-type
  h2.p-accordion__heading > .p-accordion__tab,
  p-accordion__heading.p-heading--2 > .p-accordion__tab {
    $icon-top-position: calc(#{$table-cell-vertical-padding + (map-get($line-heights, h2) - $icon-size) * 0.5});

    background-position-y: $icon-top-position;
  }

  h3.p-accordion__heading > .p-accordion__tab,
  .p-accordion__heading.p-heading--3 > .p-accordion__tab {
    $icon-top-position: calc(#{$table-cell-vertical-padding + (map-get($line-heights, h3) - $icon-size) * 0.5});

    background-position-y: $icon-top-position;
  }

  h4.p-accordion__heading > .p-accordion__tab,
  .p-accordion__heading.p-heading--4 > .p-accordion__tab {
    $icon-top-position: calc(#{$table-cell-vertical-padding + (map-get($line-heights, h4) - $icon-size) * 0.5});

    background-position-y: $icon-top-position;
  }
  // stylelint-enable selector-max-type
```
would make it much easier to read and understand in the future.